custom_model_runner/datarobot_drum/drum/root_predictors/prediction_server.py

```diff
@@ -212,6 +209,23 @@ def chat():
 
             return response, response_status
 
+        @model_api.route("/directAccess/<path:path>", methods=["GET", "POST", "PUT"])
+        @model_api.route("/nim/<path:path>", methods=["GET", "POST", "PUT"])
+        def forward_request(path):
+            openai_host = os.environ.get("OPENAI_HOST", "localhost")
+            openai_port = os.environ.get("OPENAI_PORT", "8000")
+
+            resp = requests.request(
```

Comment:
using blocking `requests` lib here, since we're using Flask already which is also blocking.

I'll do performance testing with `CUSTOM_MODEL_WORKERS=10`. 
After 11.0 we're going to either:
-  use asyncio/FastAPI for proxying such requests + we need to implement gRPC proxy as well.
- run some NIM models in a single-container mode, so we don't need to proxy requests. This will be possible only if/when NVIDIA adds a couple of changes to their NIM Containers. e.g. ability to set the `--root-path` from env variables.

Reply:
I think `CUSTOM_MODEL_WORKERS=10` (or higher) is a fine solution for the foreseeable future. Like how many concurrent requests can a NIM actually handle before we need to start looking into spinning up more replicas? In other words, is DRUM's sync nature the limiting factor even if `CUSTOM_MODEL_WORKERS=40` (max supported value).

I agree that some day it will be nicer if we switched to an asyncio HTTP framework since that will simplify not having to worry about IPC.

Reply:
> Like how many concurrent requests can a NIM actually handle before we need to start looking into spinning up more replicas?

3 concurrent requests per second for `meta/llama-3.1-8b-instruct:1.2.2` https://docs.google.com/document/d/1SO4emK8HNtZjl0obuUby3YEyDR2DMJhbp_jsJsdYpU0/edit?tab=t.0#bookmark=id.cwmozxrjpxqs